### ISSUE TITLE
Diffusion weight prediction: training script + ONNX export

### ISSUE THREAD

**Issue by TWoolff** [TWoolff/strata-training-data#203]
## Problem/Feature

Training loop and ONNX export for the diffusion weight prediction model built in issue #202. This model takes dual inputs (31-dim vertex features + segmentation encoder features) and predicts semantically-aware skinning weights.

**Depends on:** #202 (architecture + dataset loader)

## Affected Modules

- `training/train_diffusion_weights.py` — new training script
- `training/export_onnx.py` — add `diffusion_weights` model type
- `training/evaluate.py` — add evaluation support

### Training Script (`training/train_diffusion_weights.py`)

Follow the pattern in `train_weights.py` but with dual-input forward pass:

```python
# Forward pass
features = batch["features"]           # [B, 31, N, 1]
diffusion_features = batch["diffusion_features"]  # [B, C, N, 1]
outputs = model(features, diffusion_features)
```

**Loss function:** Same as standard weight model:
- KL divergence for soft weight labels (weight=1.0)
- BCE for per-vertex confidence (weight=0.5)

**CLI:**
```bash
python training/train_diffusion_weights.py --config training/configs/diffusion_weights.yaml
```

### ONNX Export

Add `diffusion_weights` variant to `export_onnx.py`:

**ONNX contract** (must match `src-tauri/src/ai/weights.rs` → `infer_dual_input()`):
- Input names: `"features"` `[1, 31, 2048, 1]`, `"diffusion_features"` `[1, C, 2048, 1]`
- Output names: `"weights"` `[1, 20, 2048, 1]`, `"confidence"` `[1, 1, 2048, 1]`
- Output filename: `diffusion_weight_prediction.onnx`

The wrapper class needs to handle dual inputs:
```python
class DiffusionWeightONNXWrapper(nn.Module):
    def forward(self, features, diffusion_features):
        out = self.model(features, diffusion_features)
        return out["weights"], out["confidence"]
```

**Export command:**
```bash
python training/export_onnx.py --model diffusion_weights \
    --checkpoint checkpoints/diffusion_weights/best.pt \
    --output models/diffusion_weight_prediction.onnx
```

### Evaluation

Add to `evaluate.py`:
- Compare against standard weight model as baseline (MAE delta)
- Per-bone MAE breakdown
- Identify where encoder features help most (e.g., shoulder/hip regions on chibi characters)
- Generate comparison visualizations

### Validation Against Baseline

After training, compare diffusion weight model vs standard weight model:
```bash
python training/evaluate.py --model diffusion_weights \
    --checkpoint checkpoints/diffusion_weights/best.pt \
    --config training/configs/diffusion_weights.yaml \
    --baseline-checkpoint checkpoints/weights/best.pt \
    --output-dir evaluation/diffusion_weights/
```

## Acceptance Criteria

- [ ] `training/train_diffusion_weights.py` trains to completion
- [ ] TensorBoard logging works: `tensorboard --logdir checkpoints/diffusion_weights/`
- [ ] Checkpoint save/load/resume works correctly
- [ ] Early stopping triggers on `val/mae`
- [ ] `export_onnx.py --model diffusion_weights` produces valid `diffusion_weight_prediction.onnx`
- [ ] ONNX validation confirms dual-input names `"features"` + `"diffusion_features"` and output names `"weights"` + `"confidence"`
- [ ] `evaluate.py` generates comparison metrics against standard weight model baseline
- [ ] Tests pass: `python -m pytest tests/ -v`
- [ ] Lint passes: `ruff check . && ruff format --check .`

## Code Standards

- Type hints on function signatures
- `snake_case` for functions/variables, `ALL_CAPS` for constants
- `pathlib.Path` for file paths
- Google-style docstrings for public functions
- `from __future__ import annotations` at top of each module
- Follow existing patterns in `train_weights.py` and `export_onnx.py`

**Comment by TWoolff** [comment]
Already done — diffusion weight prediction trained on first A100 run (0.089 MAE). Pipeline exists in training/train_diffusion_weights.py.
